terraform/modules/bucket/main.tf

```diff
@@ -88,7 +92,7 @@ resource "aws_s3_bucket_server_side_encryption_configuration" "this" {
 
   rule {
     apply_server_side_encryption_by_default {
-      kms_master_key_id = module.bucket_key.id
+      kms_master_key_id = data.aws_kms_alias.kms_key.target_key_arn
```

Comment:
Yes makes sense to switch this to the shared KMS key while keeping the module block above to allow for us to re-encrypt with the new default before removing the bucket keys in a future PR.